torch/csrc/cuda/memory_snapshot.cpp

```diff
@@ -105,16 +105,16 @@ void _initRecordAnnotations() {
         at::RecordFunctionCallback(
             [](const at::RecordFunction& fn)
                 -> std::unique_ptr<at::ObserverContext> {
-              unwind::Frame frame{fn.name(), "START", 0};
               auto r = std::make_shared<CapturedTraceback>();
-              r->recordUserDefinedFrame(frame);
+              r->recordUserDefinedMetadata("name", fn.name());
```

Comment:
I think it would be possible to simplify here by making the user-defined metadata a different class compared to trace entry. In particular, it could just have a time stamp, and then a str -> str dictionary, which would allow all this information to go in one entry.

Reply:
This is a great point, I've changed to a different struct for AnnotationEntry, which will only hold what it needs.